handsontable/src/plugins/copyPaste/actionInfo.js

```diff
@@ -0,0 +1,446 @@
+import { normalizeRanges } from './copyableRanges';
+import { isDefined } from '../../helpers/mixed';
+import { toSingleLine } from '../../helpers/templateLiteralTag';
+import { warn } from '../../helpers/console';
+import {
+  getDataByCoords,
+  getDataWithHeadersByConfig,
+  getHTMLByCoords,
+  getHTMLFromConfig,
+  htmlToGridSettings,
+} from '../../utils/parseTable';
+import { parse } from '../../3rdparty/SheetClip';
+
+const META_HEAD = [
+  '<meta name="generator" content="Handsontable"/>',
+  '<style type="text/css">td{white-space:normal}br{mso-data-placement:same-cell}</style>',
+].join('');
+
+/**
+ * Creates an object containing information about performed action: copy, cut (performing also copying) or paste.
+ *
+ * @private
+ */
+export class ActionInfo {
+  #html;
+  /**
+   * Copied data stored as array of arrays.
+   *
+   * @type {string[][]}
+   */
+  #data;
+  /**
+   * @param {object} config Configuration object for the action.
+   * @param {'copy'|'paste'} config.type Type of the action - copying (and cutting) or pasting the data.
+   * @param {string} config.html Sanitized data of "text/html" type inside the clipboard.
+   * @param {Core} [config.instance] Handsontable instance (used only while copying data).
```

Comment:
I would avoid the word "instance". We made the same mistake in Walkontable and to this day I often don't know what "instance" means. Is it the Handsontable instance or Walkontable or yet another object.

I'd recommend `hot` or `hotInstance`?